### PR TITLE
cleanup(snippet-bot): factor out the core logic

### LINKED ISSUE #1081: refactor(snippet-bot): factor out the core logic
Original suggestion:
https://github.com/googleapis/repo-automation-bots/pull/1080#discussion_r518264036

### PR DESCRIPTION
**Pull request by tmatsuo**
fixes #1081

### CODE REVIEW

File: packages/snippet-bot/src/snippet-bot.ts [hunk 1]
```diff
@@ -143,376 +143,392 @@ async function getFiles(dir: string, allFiles: string[]) {
...
+async function fullScan(context: Context, configuration: Configuration) {
+  const installationId = context.payload.installation.id;
+  const commentMark = `<!-- probot comment [${installationId}]-->`;
+  const owner = context.payload.repository.owner.login;
+  const repo = context.payload.repository.name;
+  const defaultBranch = context.payload.repository.default_branch;
+
+  if (context.payload.issue?.title.includes(FULL_SCAN_ISSUE_TITLE)) {
+    // full scan start
+    const issueNumber = context.payload.issue.number;
```
Comment (tbpg) [inline packages/snippet-bot/src/snippet-bot.ts:170]: Consider inverting this check and returning early, so you can get rid of one level of indentation.

```suggestion
  if (!context.payload.issue?.title.includes(FULL_SCAN_ISSUE_TITLE)) {
    return;
  }
  const issueNumber = context.payload.issue.number;
  // ...
```

**Review by tbpg — CHANGES_REQUESTED** [review]
One control flow comment. Otherwise, LGTM. I didn't look into logic too much, assuming it's mostly moving existing code around.

File: packages/snippet-bot/src/snippet-bot.ts (hunk 1 above)
Comment (tmatsuo) [reply packages/snippet-bot/src/snippet-bot.ts:170]: Good suggestion. Will do.

Comment (tmatsuo) [reply packages/snippet-bot/src/snippet-bot.ts]: Done

**Comment by tmatsuo** [comment]
@tbpg Yes, mostly just moving around the code. PTAL

**Review by tbpg — APPROVED** [review]

**Pull request merged**